File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/trace/unresolved/UnresolvedMetaReference.java
package org.kevoree.modeling.api.trace.unresolved;

import org.kevoree.modeling.api.meta.MetaClass;
import org.kevoree.modeling.api.meta.MetaReference;

/**
 * Created by duke on 10/16/14.
 */
public class UnresolvedMetaReference implements MetaReference {

    public UnresolvedMetaReference(String p_metaName) {
        this._metaName = p_metaName;
    }

    private String _metaName;

    @Override
    public boolean contained() {
        return false;
    }

    @Override
    public boolean single() {
        return false;
    }

    @Override
    public MetaClass metaType() {
        return null;
    }

    @Override
    public MetaReference opposite() {
        return null;
    }

    @Override
    public MetaClass origin() {
        return null;
    }

    @Override
    public String metaName() {
        return _metaName;
    }

    @Override
    public int index() {
        return -1;
    }
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/trace/unresolved/UnresolvedMetaReference.ts
///<reference path="../../meta/MetaClass.ts"/>
///<reference path="../../meta/MetaReference.ts"/>

class UnresolvedMetaReference implements MetaReference {

  private _metaName: string = null;

  constructor(p_metaName: string) {
    this._metaName = p_metaName;
  }

  public contained(): boolean {
    return false;
  }

  public single(): boolean {
    return false;
  }

  public metaType(): MetaClass {
    return null;
  }

  public opposite(): MetaReference {
    return null;
  }

  public origin(): MetaClass {
    return null;
  }

  public metaName(): string {
    return this._metaName;
  }

  public index(): number {
    return -1;
  }

}

